src/DynamoCore/Models/NodeModel.cs

```diff
@@ -1746,6 +1747,22 @@ public NodeCategoryAttribute(string category)
     [AttributeUsage(AttributeTargets.All)]
     public class NodeSearchTagsAttribute : Attribute
     {
+        public NodeSearchTagsAttribute(string tagsID, Type resourceType)
+        {
+            if (resourceType == null)
+                throw new ArgumentNullException("resourceType");
+
+            var prop = resourceType.GetProperty(tagsID, BindingFlags.Public | BindingFlags.Static);
+            if (prop != null && prop.PropertyType == typeof(String))
+            {
+                var tagString = (string)prop.GetValue(null, null);
```

Comment:
Convince me that this is never going to throw a class cast so we don't need to do an x as y thing?